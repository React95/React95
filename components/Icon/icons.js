import arrow_left from '../shared/assets/icons/arrow_left.ico';
import arrow_right from '../shared/assets/icons/arrow_right.ico';
import bat from '../shared/assets/icons/bat.ico';
import bat_exec from '../shared/assets/icons/bat_exec.ico';
import bat_exec_2 from '../shared/assets/icons/bat_exec_2.ico';
import bat_wait from '../shared/assets/icons/bat_wait.ico';
import bookmark from '../shared/assets/icons/bookmark.ico';
import brush from '../shared/assets/icons/brush.ico';
import bulb from '../shared/assets/icons/bulb.ico';
import calculator from '../shared/assets/icons/calculator.ico';
import camera from '../shared/assets/icons/camera.ico';
import computer from '../shared/assets/icons/computer.ico';
import computer_2 from '../shared/assets/icons/computer_2.ico';
import computer_3 from '../shared/assets/icons/computer_3.ico';
import computer_4 from '../shared/assets/icons/computer_4.ico';
import computer_find from '../shared/assets/icons/computer_find.ico';
import defrag from '../shared/assets/icons/defrag.ico';
import desktop from '../shared/assets/icons/desktop.ico';
import earth from '../shared/assets/icons/earth.ico';
import explore from '../shared/assets/icons/explore.ico';
import file_delete from '../shared/assets/icons/file_delete.ico';
import file_find from '../shared/assets/icons/file_find.ico';
import file_find2 from '../shared/assets/icons/file_find2.ico';
import file_find3 from '../shared/assets/icons/file_find3.ico';
import file_corrupted from '../shared/assets/icons/file_corrupted.ico';
import file_font from '../shared/assets/icons/file_font.ico';
import file_font_2 from '../shared/assets/icons/file_font_2.ico';
import file_pen from '../shared/assets/icons/file_pen.ico';
import file_pencil from '../shared/assets/icons/file_pencil.ico';
import file_pin from '../shared/assets/icons/file_pin.ico';
import file_settings from '../shared/assets/icons/file_settings.ico';
import file_text from '../shared/assets/icons/file_text.ico';
import file_text_settings from '../shared/assets/icons/file_text_settings.ico';
import file_transfer from '../shared/assets/icons/file_transfer.ico';
import files from '../shared/assets/icons/files.ico';
import folder from '../shared/assets/icons/folder.ico';
import folder_exe from '../shared/assets/icons/folder_exe.ico';
import folder_exe2 from '../shared/assets/icons/folder_exe2.ico';
import folder_file from '../shared/assets/icons/folder_file.ico';
import folder_font from '../shared/assets/icons/folder_font.ico';
import folder_open from '../shared/assets/icons/folder_open.ico';
import folder_print from '../shared/assets/icons/folder_print.ico';
import folder_rename from '../shared/assets/icons/folder_rename.ico';
import folder_settings from '../shared/assets/icons/folder_settings.ico';
import folder_settings_2 from '../shared/assets/icons/folder_settings_2.ico';
import folder_shared from '../shared/assets/icons/folder_shared.ico';
import forbidden from '../shared/assets/icons/forbidden.ico';
import globe from '../shared/assets/icons/globe.ico';
import hand from '../shared/assets/icons/hand.ico';
import help_book from '../shared/assets/icons/help_book.ico';
import issue from '../shared/assets/icons/issue.ico';
import install from '../shared/assets/icons/install.ico';
import key from '../shared/assets/icons/key.ico';
import keys from '../shared/assets/icons/keys.ico';
import loader_bat from '../shared/assets/icons/loader_bat.ico';
import lock from '../shared/assets/icons/lock.ico';
import logo from '../shared/assets/icons/logo.ico';
import mail from '../shared/assets/icons/mail.ico';
import media_cd from '../shared/assets/icons/media_cd.ico';
import memory from '../shared/assets/icons/memory.ico';
import mic from '../shared/assets/icons/mic.ico';
import mute from '../shared/assets/icons/mute.ico';
import network from '../shared/assets/icons/network.ico';
import network_2 from '../shared/assets/icons/network_2.ico';
import network_3 from '../shared/assets/icons/network_3.ico';
import notepad from '../shared/assets/icons/notepad.ico';
import packager from '../shared/assets/icons/packager.ico';
import phone from '../shared/assets/icons/phone.ico';
import plugin from '../shared/assets/icons/plugin.ico';
import power_off from '../shared/assets/icons/power_off.ico';
import power_on from '../shared/assets/icons/power_on.ico';
import printer from '../shared/assets/icons/printer.ico';
import printer_calendar from '../shared/assets/icons/printer_calendar.ico';
import printer_drive from '../shared/assets/icons/printer_drive.ico';
import printer_shared from '../shared/assets/icons/printer_shared.ico';
import reader_cd from '../shared/assets/icons/reader_cd.ico';
import reader_cd_2 from '../shared/assets/icons/reader_cd_2.ico';
import reader_closed from '../shared/assets/icons/reader_closed.ico';
import reader_disket from '../shared/assets/icons/reader_disket.ico';
import reader_disket_casset from '../shared/assets/icons/reader_disket_casset.ico';
import reader_disket2 from '../shared/assets/icons/reader_disket2.ico';
import reader_eject from '../shared/assets/icons/reader_eject.ico';
import reader_noshared from '../shared/assets/icons/reader_noshared.ico';
import reader_opened from '../shared/assets/icons/reader_opened.ico';
import reader_shared from '../shared/assets/icons/reader_shared.ico';
import regedit from '../shared/assets/icons/regedit.ico';
import recycle_empty from '../shared/assets/icons/recycle_empty.ico';
import recycle_file from '../shared/assets/icons/recycle_file.ico';
import recycle_filefolder from '../shared/assets/icons/recycle_filefolder.ico';
import recycle_folder from '../shared/assets/icons/recycle_folder.ico';
import recycle_full from '../shared/assets/icons/recycle_full.ico';
import settings from '../shared/assets/icons/settings.ico';
import shortcut from '../shared/assets/icons/shortcut.ico';
import shortcut2 from '../shared/assets/icons/shortcut2.ico';
import star from '../shared/assets/icons/star.ico';
import textchat from '../shared/assets/icons/textchat.ico';
import textchat_2 from '../shared/assets/icons/textchat_2.ico';
import timedate from '../shared/assets/icons/timedate.ico';
import tree from '../shared/assets/icons/tree.ico';
import unmute from '../shared/assets/icons/unmute.ico';
import user from '../shared/assets/icons/user.ico';
import warning from '../shared/assets/icons/warning.ico';
import wordpad from '../shared/assets/icons/wordpad.ico';
import windows_explorer from '../shared/assets/icons/windows_explorer.ico';
import ms_dos from '../shared/assets/icons/ms_dos.ico';
import microsoft_network from '../shared/assets/icons/microsoft_network.ico';
import microsoft_exchange from '../shared/assets/icons/microsoft_exchange.ico';

const path = {
  arrow_left,
  arrow_right,
  bat,
  bat_exec,
  bat_exec_2,
  bat_wait,
  bookmark,
  brush,
  bulb,
  calculator,
  camera,
  computer,
  computer_2,
  computer_3,
  computer_4,
  computer_find,
  defrag,
  desktop,
  earth,
  explore,
  file_delete,
  file_find,
  file_find2,
  file_find3,
  file_corrupted,
  file_font,
  file_font_2,
  file_pen,
  file_pencil,
  file_pin,
  file_settings,
  file_text,
  file_text_settings,
  file_transfer,
  files,
  folder,
  folder_exe,
  folder_exe2,
  folder_file,
  folder_font,
  folder_open,
  folder_print,
  folder_rename,
  folder_settings,
  folder_settings_2,
  folder_shared,
  forbidden,
  globe,
  hand,
  help_book,
  install,
  issue,
  key,
  keys,
  loader_bat,
  lock,
  logo,
  mail,
  media_cd,
  memory,
  mic,
  mute,
  network,
  network_2,
  network_3,
  notepad,
  packager,
  phone,
  plugin,
  power_off,
  power_on,
  printer,
  printer_calendar,
  printer_drive,
  printer_shared,
  reader_cd,
  reader_cd_2,
  reader_closed,
  reader_disket,
  reader_disket_casset,
  reader_disket2,
  reader_eject,
  reader_noshared,
  reader_opened,
  reader_shared,
  recycle_empty,
  recycle_file,
  recycle_filefolder,
  recycle_folder,
  recycle_full,
  regedit,
  settings,
  star,
  shortcut,
  shortcut2,
  textchat,
  textchat_2,
  timedate,
  tree,
  unmute,
  user,
  wordpad,
  warning,
  windows_explorer,
  ms_dos,
  microsoft_network,
  microsoft_exchange,
};

export default path;
