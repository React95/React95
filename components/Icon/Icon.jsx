import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import icons from './icons.js';

const size = ({ width, height }) => css`
    width: ${width}px;
    height: ${height}px;
  `;

const I = styled.i`
  display: block;
  ${size}

  background-image: url('${({ name }) => icons[name] || 'none'}');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
`;

const Icon = ({
  name, width, height, ...rest
}) => (
  <I name={name} width={width} height={height} {...rest} />
);

Icon.propTypes = {
  name: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
};

Icon.defaultProps = {
  name: null,
  width: 32,
  height: 32,
};

export default Icon;
