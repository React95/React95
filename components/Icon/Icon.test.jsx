import React from 'react';
import Icon from './Icon';
import { render } from 'react-testing-library';

describe('<Icon />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Icon name="bat" />);
      expect(container).toMatchSnapshot();
    });
  });
});
