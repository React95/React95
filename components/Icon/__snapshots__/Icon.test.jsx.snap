// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Snapshots should match snapshot 1`] = `
.c0 {
  display: block;
  width: 32px;
  height: 32px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

<div>
  <i
    class="c0"
    height="32"
    name="bat"
    width="32"
  />
</div>
`;
